rest/server_context.go

```diff
@@ -999,6 +1006,9 @@ func (sc *ServerContext) stopStatsLogger() {
 	if sc.statsContext.statsLoggingTicker != nil {
 		sc.statsContext.statsLoggingTicker.Stop()
 	}
+	if sc.statsContext.statsLoggingStop != nil {
+		close(sc.statsContext.statsLoggingStop)
+	}
```

Comment:
I think the close channel is a reasonable approach, but stopStatsLogger shouldn't need to stop both the ticker and the terminator channel.  You can move the ticker.Stop after line 995.